skyportal/handlers/api/galaxy.py

```diff
@@ -0,0 +1,109 @@
+from baselayer.app.access import permissions, auth_or_token
+
+import astropy.units as u
+import healpix_alchemy as ha
+
+from ..base import BaseHandler
+from ...models import DBSession, Galaxy
+
+
+class GalaxyCatalogHandler(BaseHandler):
+    @permissions(['System admin'])
+    async def post(self):
+        """
+        ---
+        description: Ingest a Galaxy catalog
+        tags:
+          - galaxies
+        requestBody:
+          content:
+            application/json:
+              schema: GalaxyHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        catalog_name = data.get('catalog_name')
+        catalog_data = data.get('catalog_data')
+
+        if catalog_name is None:
+            return self.error("catalog_name is a required parameter.")
+        if catalog_data is None:
+            return self.error("catalog_data is a required parameter.")
+
+        galaxies = [
```

Comment:
There's no enforcement of the data structure of the posted json. That's needed. Also bounds checking is needed otherwise we could get nonsensical values added to the DB. This is the place to ensure proper values in the DB. Could defer the checking to the Galaxy post and catch errors here instead.

Reply:
So we could enforce that all distmpc, distmpc_unc, redshift, and redshift_error are all > 0. I guess the minimum viable product allowed should be ra, dec, and name.

Reply:
Ok I added some checking along these lines.